30-performance/TPC-DS Runner/README.md

```diff
@@ -0,0 +1,57 @@
+## Contributing
+#### Backlog
+Required
+* Support query duplication. Currently, beaker does not expose the ability to repeatedly run the same query so we just rerun the benchmark. This makes each benchmark run blocking.
+  * Once merged, the loop logic will need to be changed also
+* Check if there is a warehouse continuation token - I didn't see it in the API docs
+* Change dist to reference main instead of nishant branch of beaker. We're waiting on PR merge for this
+* Finish the README
+  * Quickstart
+  * Education about concurrency testing
+  * Product notes/limitations
+
+Nice to have
+* Improve token authentication. Currently the user has to pass a PAT via a filter. This is not acceptable because the PAT is exposed via the job UI. Potential solutions include but are not limited to...
+  1. Add secrets API functionality (instructions for how to pass token, add secret scope/key widgets, pass values as job parameters, read accordingly)
+  2. Refactor beaker to support OAuth - [datbaricks-sql-connector example](https://github.com/databricks/databricks-sql-python/blob/b894605ba7fe2525f1cac830a330149b09b1d8c1/examples/interactive_oauth.py). Note that you'll have to migrate all `requests` API calls to the python SDK as well
+  * [Relevant slack thread](https://databricks.slack.com/archives/C0J7QJHUJ/p1691539325013799)
+* Support writing raw queries to UC volumes instead of DBFS
+* determine if spark-sql-perf supports latest LTS DBR version or if we need to hardcode 12.2
+* Make Beaker pip-installable within a Databricks notebook - [issue](https://github.com/goodwillpunning/beaker/issues/19)
+* Add dashboarding and further analysis using Nishant's tool(s)
+
+Needs Scoping
+* Leverage beaker warehouse creation instead of current method
+
+# Databricks TPC-DS Benchmarking Tool
+
+This tool runs the TPC-DS benchmark on a Databricks SQL warehouse. The TPC-DS benchmark is a standardised method of evaluating the performance of decision support solutions, such as databases, data warehouses, and big data systems.
+
```

Comment:
I want to know what will happen. Maybe this will be in the README, but I think the README has good higher level information. So maybe something like below should be here.

----
This script submits a job (in Databricks workflows) that does the work. The job has two tasks:
 - Task 1: Create the TPC-DS dataset of your [chosen size] in the [catalog.schema].
 - Task 2: Create a dbsql warehouse of [cluster_size] and [max_clusters] and run the TPC-DS queries on top of the created dataset with [concurrency]. Create a results table (if it does :)).

Reply:
Added